source/main.c

```diff
@@ -256,9 +256,21 @@ int load_usb(char slot)
     data = GC_READY;
     usb_sendbuffer_safe(channel, &data, 1);
 
-    kprintf("Waiting for ack...\n");
+    kprintf("Waiting for ack (5s timeout)...\n");
+    u64 current_time, start_time = gettime();
+
     while ((data != PC_READY) && (data != PC_OK))
-        usb_recvbuffer_safe(channel, &data, 1);
+    {
+        current_time = gettime();
+
```

Comment:
either remove the blank line (this is directly relevant to the if condition below) or fold the `gettime()` call into it, that shouldn't hurt readability imo